app/src/main/java/app/revanced/manager/ui/component/InstallerStatusDialog.kt

```diff
@@ -0,0 +1,158 @@
+package app.revanced.manager.ui.component
+
+import android.content.pm.PackageInstaller
+import androidx.annotation.RequiresApi
+import androidx.annotation.StringRes
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Check
+import androidx.compose.material.icons.outlined.ErrorOutline
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import app.revanced.manager.R
+import app.revanced.manager.ui.viewmodel.PackageInstallerResult
+import com.github.materiiapps.enumutil.FromValue
+
+internal abstract class InstallerStatusDialogModel {
+    abstract var packageInstallerResult: PackageInstallerResult
+    abstract var showInstallerStatusDialog: Boolean
+
+    abstract fun reinstall()
+    abstract fun install()
+}
+
+private typealias InstallerStatusDialogButtonHandler = ((model: InstallerStatusDialogModel) -> Unit)
+private typealias InstallerStatusDialogButton = @Composable (model: InstallerStatusDialogModel) -> Unit
+
+
+@Composable
+internal fun InstallerStatusDialog(
```

Comment:
```suggestion
fun InstallerStatusDialog(
```
Is internal necessary?